pkg/util/tpm/tpm.go

```diff
@@ -0,0 +1,424 @@
+package tpm
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-tspi/tpmclient"
+	"github.com/coreos/go-tspi/tspiconst"
+	"github.com/coreos/go-tspi/verification"
+	"github.com/golang/glog"
+	"github.com/mitchellh/mapstructure"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/typed/dynamic"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type TPMHandler struct {
+	tpmclient    *dynamic.ResourceClient
+	PolicyClient *dynamic.ResourceClient
+}
+
+var TPMResource string = "coreos.com"
+
+type Tpm struct {
+	// The TPM's EK certificate
+	EKCert []byte
+	// The encrypted AIK keyblob
+	AIKBlob []byte
+	// The public half of AIK
+	AIKPub []byte
+	// The current address associated with the TPM
+	Address string
+}
+
+func (t *TPMHandler) Setup(config *restclient.Config) error {
+	if err := client.SetKubernetesDefaults(config); err != nil {
+		return err
+	}
+	config.APIPath = "apis/coreos.com"
```

Comment:
do we want this versioned? apis/alpha.coreos.com? Also would prefer that the string is added as a const